pennylane/debugging.py

```diff
@@ -53,21 +53,61 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
+    r"""Transforms a QNode into several tapes by aggregating all operations up to a `qml.Snapshot`
+    operation into their own execution tape.
+
+    The transform is conservative about the wires that it includes in each tape.
+    So, if all operations preceding a snapshot in a 3-qubit circuit has been applied to only one wire,
+    the tape would only be looking at this wire. This can be overriden by the configuration of the execution device
+    and its nature. For supported simulators, measurements with no specified wires would default to all wires.
+
+    The output is a dictionary where each key is either the tag supplied to the snapshot or its
+    index in order of appearance, in additition to an "execution_results" that returns the final output
+    of the quantum circuit. The post-processing function is responsible for aggregating the results into
+    this dictionary.
+
+    When the transform is applied to a QNode, the `shots` configuration would be inherited by the `Snapshot`
+    instances, as long as the chosen measurement supports the provided value and is permitted by the device.
+
+    It should be noted that for unsupported devices, a separate execution would be carried out for each snapshot.
+    As illustrated in one of the examples below.
 
     Args:
-        qnode (.QNode): the input QNode to be simulated
+        tape (QNode or QuantumTape or Callable): a quantum circuit.
 
     Returns:
-        A function that has the same argument signature as ``qnode`` and returns a dictionary.
-        When called, the function will execute the QNode on the registered device and retrieve
-        the saved snapshots obtained via the :class:`~.pennylane.Snapshot` operation. Additionally,
-        the snapshot dictionary always contains the execution results of the QNode, so the use of
-        the tag "execution_results" should be avoided to prevent conflicting key names.
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumTape], function]: The
+        transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    .. warning::
+
+        For devices that do not support snapshots (e.g QPUs, external plug-in simulators), be mindful of
+        additional costs that you might incur due to the 1 separate execution/snapshot behaviour.
 
     **Example**
 
+    .. code-block:: python3
+
+        ops = [
+            qml.Snapshot(),
+            qml.Hadamard(wires=0),
+            qml.Snapshot("very_important_state"),
+            qml.CNOT(wires=[0, 1]),
+            qml.Snapshot(),
+        ]
+
+        measurements = [qml.expval(qml.PauliX(0))]
+
+        tape = qml.tape.QuantumTape(ops, measurements)
+
+        tapes, collect_results_into_dict = qml.snapshots(tape)
+
+    >>> print(tapes)
+    [<QuantumTape: wires=[], params=0>, <QuantumTape: wires=[0], params=0>, <QuantumTape: wires=[0, 1], params=0>, <QuantumTape: wires=[0, 1], params=0>]
+
+
```

Comment:
The "developer UI example" should probably be placed below the recommended user interface.  The qnode examples should be the first thing a user sees when they are trying to figure out how they should use `snapshots`.